src/commands/dev/lib/default-should-serve.ts

```diff
@@ -0,0 +1,29 @@
+import { parse, basename, extname } from 'path';
+// @ts-ignore
+import { ShouldServeParams } from '@now/build-utils';
+
+export default function shouldServe({
+  entrypoint,
+  files,
+  requestPath
+}: ShouldServeParams): boolean {
+  requestPath = requestPath.replace(/\/$/, '');
+
+  if (entrypoint === requestPath) {
+    return true;
+  }
+
+  if (isIndex(entrypoint)) {
+    if (parse(entrypoint).dir === requestPath) {
```

Comment:
```suggestion
    if (dirname(entrypoint) === requestPath) {
```

Reply:
Use dirname would cause `path.dirname('') === '.'`

Reply:
Strange!

Maybe you could change the code to do `files = parse(entrypoint)` once and then compare `file.name === 'index'` and `file.dir === requestPath`?

Reply:
I think this comparison will fail on Windows because the path separator is different.

See [path](https://nodejs.org/api/path.html#path_path_parse_path) docs for an example.

Reply:
Nice!

Reply:
@styfle Made a fix by my guess, haven't tested https://github.com/zeit/now-cli/pull/2150/commits/4d365ac691c661ef9a0704524b34a6f90ad15ad5